spec/models/book_spec.rb

```diff
@@ -19,6 +19,49 @@
     it {should have_many :reviews}
   end
 
+  describe "class methods" do
+    before(:each) do
+      @book_1 = Book.create!(title: "The Frozen Deep", page_count: 106, year_published: 1874, thumbnail: "https://images.gr-assets.com/books/1328728986l/1009218.jpg")
+      @book_2 = Book.create!(title: "To Kill a Mockingbird", page_count: 600, year_published: 1960, thumbnail: "https://upload.wikimedia.org/wikipedia/en/7/79/To_Kill_a_Mockingbird.JPG")
+      @book_3 = Book.create!(title: "50 Shades of Grey", page_count: 514, year_published: 2011, thumbnail: "https://upload.wikimedia.org/wikipedia/en/5/5e/50ShadesofGreyCoverArt.jpg")
+      @author_1 = @book_1.authors.create!(name: "Wilkie Collins")
+      @author_2 = @book_1.authors.create!(name: "Charles Dickens")
+      @author_3 = @book_2.authors.create!(name: "Harper Lee")
+      user_1 = User.create(username: "Chris Davis")
+      user_2 = User.create(username: "Alexandra Chakeres")
+      @review_1 = @book_2.reviews.create!(title: "This book rocks!", rating: 5, text: "Read it!", user: user_1)
+      @review_2 = @book_1.reviews.create!(title: "This book sucks!", rating: 1, text: "Don't read it!", user: user_2)
+      @review_3 = @book_2.reviews.create!(title: "It's OK.", rating: 3, text: "Meh", user: user_2)
+    end
+
+    it "sorts books by average rating (ascending)" do
+      expect(Book.sort_by_rating).to eq([@book_3, @book_1, @book_2])
+      expect(Book.sort_by_rating(true)).to eq([@book_3, @book_1, @book_2])
+    end
+
+    it "sorts books by average rating (descending)" do
+      expect(Book.sort_by_rating(false)).to eq([@book_2, @book_1, @book_3])
+    end
+
+    it "sorts books by number of pages (ascending)" do
+      expect(Book.sort_by_page_count).to eq([@book_1, @book_3, @book_2])
+      expect(Book.sort_by_page_count(true)).to eq([@book_1, @book_3, @book_2])
+    end
+
+    it "sorts books by number of pages (descending)" do
+      expect(Book.sort_by_page_count(false)).to eq([@book_2, @book_3, @book_1])
+    end
+
+    it "sorts books by number of reviews (ascending)" do
+      expect(Book.sort_by_review_count).to eq([@book_3, @book_1, @book_2])
+      expect(Book.sort_by_review_count(true)).to eq([@book_3, @book_1, @book_2])
+    end
+
+    it "sorts books by number of reviews (descending)" do
+      expect(Book.sort_by_review_count(false)).to eq([@book_2, @book_1, @book_3])
+    end
+  end
+
```

Comment:
The only sad path testing I can think of is a manual query that's malformed, visit "/books_path?sort=thisisbroken. It should just return Book.all as usual but we don't have an explicit test for this.

Reply:
Good one -- added that test (to the feature test, since it has to do with the URL)